changes/next/cyr-1355-remove-xstats

```diff
@@ -0,0 +1,24 @@
+Description:
+
+The undocumented XSTATS metrics system has been removed.
+
+
+Documentation:
+
+There is none.
+
+
+Config changes:
+
+None.
+
+
+Upgrade instructions:
+
+If you somehow had found XSTATS and were using it, stop!  Look at the
+Prometheus metrics instead.
+
+
+GitHub issue:
+
+https://github.com/cyrusimap/cyrus-imapd/pull/5659
```

Comment:
FYI This field doesn't need to be self-referential, it's just if there was an actual github issue that this PR addresses (so we can write "Fixed #xxxx: blah" instead of just "Fixed: blah" or whatever).  Or I guess if we're retroactively adding a changes file for a previous change that was lacking one.  But usually the PR number will be in the merge commit that added the changes file, if we need it, so it doesn't need to be here too.